### DISCUSSION TITLE
Deploy Flutter web with terraform from codemagic

### DISCUSSION THREAD

**Discussion by kgaulin** [codemagic-ci-cd/codemagic-docs#1313] [Q&A] [answered]
Hi folks, 

i have a question about deploiement. Can i use terraform with Post-publish script ?

**Comment by himesh-cm** [comment, marked as answer]
Hi @kgaulin, I suggest referring to this [documentation](https://docs.codemagic.io/yaml-publishing/cloudflare-pages/) for publishing to Cloudflare, which should give you an idea of how you can set the pipeline with terraform; I have personally not tried, but theoretically, it should be possible to achieve.

Do let us know how it goes.

**Reply by himesh-cm** [reply]
Hi @kgaulin, please share more context on what you are trying to achieve with terraform so that I can assist you better.

**Reply by kgaulin** [reply]
hi @himesh-cm, we use terraform to manage aws infrastructure and push new build to aws s3 for all your webapp using gitlab ci. We just begin to write a flutter application and we want to manage the aws s3 and cloudfront with terraform like other web app. I would like to use only codemagic to do it and keep all build of the flutter app at the same place.

**Reply by himesh-cm** [reply]
Hi @kgaulin, thanks for sharing the use-case, kindly allow me sometime to get back to you.

**Reply by himesh-cm** [reply]
Hi @kgaulin,

You can set up the script to install terraform with homebrew and the script that specifies the configuration to create resources on AWS in the post-publishing section of your workflow. You can also check if the build is successful and then trigger the terraform script
You can read it more about [here](https://docs.codemagic.io/yaml-publishing/post-publish/).

Sensitive credentials can be stored as env var on codemagic.

Do let me know if you hit any roadblocks and need assistance on the same.
